# Homework Problem 1
- Note the data contains many headers, quotes and headers.  These create things that are easy to idenitify. You can remove this from the documents using  data_train = fetch_20newsgroups(subset='all', remove=('headers', 'footers', 'quotes')
- What happens to the performance of the model?  Is this to be expected?  Was it a good idea?
- I used a CountVectorizer(binary=True)
- What happens to the vectors if you set binary=False?  What happens to the performance?
- Try training an svm with a linear kernel using binary vectors, count vectors, also apply a tfidf transform to the data.  Compare performance with Naive Bayes.




In [1]:
from sklearn import datasets
from sklearn.feature_extraction.text import CountVectorizer
from sklearn.naive_bayes import MultinomialNB
from sklearn.svm import SVC
from sklearn.model_selection import train_test_split
from sklearn.metrics import confusion_matrix
from sklearn.metrics import accuracy_score

# getting 20newsgroups dataset
data = datasets.fetch_20newsgroups(subset='all')

data_train_X, data_test_X = train_test_split(data.data, random_state=1 )
data_train_y, data_test_y = train_test_split(data.target, random_state=1 )

# vectorizer = CountVectorizer(binary=False)
vectorizer = CountVectorizer(binary=True)
vectors_train=vectorizer.fit_transform(data_train_X)
vectors_test=vectorizer.transform(data_test_X)

nb = MultinomialNB(alpha=.1)
nb.fit(vectors_train, data_train_y)

#print a confusion matrix and accuracy
print(confusion_matrix(nb.predict(vectors_test), data_test_y))
print(accuracy_score(nb.predict(vectors_test), data_test_y))

[[184   0   0   0   0   0   0   0   0   0   0   0   0   0   0   1   0   0
    0  17]
 [  0 220  36   5   3  22   0   3   0   0   0   7   7   2   1   2   0   0
    0   0]
 [  0   0  81   0   2   0   1   0   0   0   0   0   0   0   0   0   1   0
    0   0]
 [  0   7  97 231  10   6   8   0   0   2   0   1   9   3   1   1   0   0
    0   0]
 [  0   3   6  10 211   0   4   0   1   0   0   0   6   0   1   1   0   0
    0   0]
 [  0   4  23   1   0 194   0   0   0   0   1   2   0   0   1   0   0   0
    0   0]
 [  1   1   2   6   3   1 214   4   4   2   0   0   1   1   1   0   0   0
    0   0]
 [  1   0   3   0   0   0  10 240   1   0   0   0   2   1   0   0   0   0
    0   1]
 [  0   0   0   0   0   1   3  10 244   0   1   0   0   0   0   0   0   0
    0   0]
 [  0   0   0   0   0   0   2   0   0 221   2   0   0   0   1   0   0   0
    0   0]
 [  0   0   0   1   0   0   1   0   0   1 255   0   0   0   0   1   0   0
    0   0]
 [  0   2   5   0   1   1   2   1   0   0   0 244   1   0   0   0

In [2]:
# getting cleaned data for training
data_clean = datasets.fetch_20newsgroups(subset='all',remove=('headers','footers','quotes'))

data_clean_train_X, data_test_X = train_test_split(data_clean.data, random_state=1 )
data_clean_train_y, data_test_y = train_test_split(data_clean.target, random_state=1 )

# vect_clean = CountVectorizer(binary=False)
vect_clean = CountVectorizer(binary=True)
vect_clean_train=vect_clean.fit_transform(data_train_X)
vect_clean_test=vect_clean.transform(data_test_X)

nb = MultinomialNB(alpha=.1)
nb.fit(vect_clean_train, data_train_y)

#print a confusion matrix and accuracy
print(confusion_matrix(nb.predict(vect_clean_test), data_test_y))
print(accuracy_score(nb.predict(vect_clean_test), data_test_y))

[[138   1   2   1   1   1   0   2   3   0   3   3   0   2   2   7   4   5
    9  21]
 [  1 181  38   4   2  27   3   2   0   0   0   6   8   3   7   2   0   2
    0   0]
 [  0   0  34   0   3   0   0   0   0   0   0   0   0   0   0   0   1   0
    0   0]
 [  0  11 104 219  17   8  14   0   2   2   0   0  14   2   2   0   0   0
    0   0]
 [  0  12  14   9 180   4  10   0   2   0   1   1   8   1   1   1   0   1
    1   0]
 [  0  10  37   2   2 175   0   1   0   0   1   2   0   0   2   0   0   1
    0   0]
 [  0   1   2   4   4   2 179   6   3   4   0   0   3   2   0   0   0   0
    0   1]
 [  2   0   0   2   4   1  15 214  16   0   1   0   4   2   3   0   1   0
    1   2]
 [  3   1   4   0   0   1   1   6 191   1   3   1   2   2   0   1   2   1
    2   0]
 [  5   7   9   7   8   2   4  12   9 209  13  11   7   7  10   5   9   6
    5   5]
 [  0   1   1   0   0   0   3   0   1   1 226   0   0   0   1   0   0   0
    1   0]
 [  4   5   4   2   2   5   5   1   1   1   1 203   4   0   3   0

# RESULT
... it looks like it got worse?

* I used a CountVectorizer(binary=True)
* What happens to the vectors if you set binary=False? What happens to the performance?
* Try training an svm with a linear kernel using binary vectors, count vectors, also apply a tfidf transform to the data. Compare performance with Naive Bayes.




In [3]:
# getting cleaned data for training
data_clean = datasets.fetch_20newsgroups(subset='all',remove=('headers','footers','quotes'))

data_clean_train_X, data_test_X = train_test_split(data_clean.data, random_state=1 )
data_clean_train_y, data_test_y = train_test_split(data_clean.target, random_state=1 )

vect_false = CountVectorizer(binary=False)
vect_false_train=vect_false.fit_transform(data_train_X)
vect_false_test=vect_false.transform(data_test_X)

nb = MultinomialNB(alpha=.1)
nb.fit(vect_false_train, data_train_y)

#print a confusion matrix and accuracy
print(confusion_matrix(nb.predict(vect_false_test), data_test_y))
print(accuracy_score(nb.predict(vect_false_test), data_test_y))

[[135   0   2   1   1   0   0   3   2   1   6   2   1   1   6  10   5   7
    9  27]
 [  1 186  45   4   4  31   4   2   1   2   0   3   9   1   6   3   0   2
    0   0]
 [  0   0   8   0   2   0   0   0   0   0   0   0   0   0   0   0   1   0
    0   0]
 [  0   8 104 210  16   5  11   0   2   0   0   0  12   2   0   0   0   0
    0   0]
 [  0  10  17  12 174   4  11   0   2   0   1   1   8   1   1   1   0   0
    0   0]
 [  0  10  44   2   3 174   0   1   0   0   0   3   0   0   1   0   0   1
    0   0]
 [  0   0   2   5   4   1 174   3   2   3   0   0   1   1   0   0   0   0
    0   0]
 [  1   0   0   1   4   2  15 217  16   0   1   1   5   2   3   0   1   0
    1   2]
 [  3   1   1   0   0   1   3   9 190   1   1   1   1   2   2   1   0   0
    0   0]
 [  5   7  10   7   7   1   4  12   9 206  17  11   7   6   9   5   7   5
    5   5]
 [  0   0   1   0   0   0   1   0   1   1 222   1   0   0   0   0   0   0
    1   0]
 [  2   7  10   2   2   3   6   0   1   0   0 200   7   0   2   0

# RESULT
... it looks like it got worse again????

Let's try it on the original...

In [4]:
# getting 20newsgroups dataset
data = datasets.fetch_20newsgroups(subset='all')

data_train_X, data_test_X = train_test_split(data.data, random_state=1 )
data_train_y, data_test_y = train_test_split(data.target, random_state=1 )

vectorizer = CountVectorizer(binary=False)
vectors_train=vectorizer.fit_transform(data_train_X)
vectors_test=vectorizer.transform(data_test_X)

nb = MultinomialNB(alpha=.1)
nb.fit(vectors_train, data_train_y)

#print a confusion matrix and accuracy
print(confusion_matrix(nb.predict(vectors_test), data_test_y))
print(accuracy_score(nb.predict(vectors_test), data_test_y))

[[183   0   1   0   0   0   0   0   0   0   0   0   0   0   0   2   0   0
    0  16]
 [  0 220  43   5   7  29   1   2   0   0   0   3   7   1   2   1   0   1
    0   0]
 [  0   0  39   0   2   0   1   0   0   0   0   0   0   0   0   0   1   0
    0   0]
 [  0   7 101 232   9   5   5   0   0   0   0   0  13   2   0   1   1   0
    0   0]
 [  0   3  25  11 209   1   4   0   1   0   0   0   5   0   1   1   0   1
    0   0]
 [  0   5  30   1   0 189   0   0   0   0   0   2   0   0   0   0   0   0
    0   0]
 [  1   2   5   5   2   1 213   3   4   1   0   1   0   2   1   0   0   0
    0   0]
 [  1   0   3   1   0   0  12 241   1   0   0   0   2   0   0   1   0   0
    0   1]
 [  0   0   2   0   0   1   2  14 244   0   1   0   1   2   0   1   0   0
    0   0]
 [  0   0   0   0   0   0   2   0   0 221   4   0   0   0   2   0   0   0
    0   1]
 [  0   0   0   1   0   0   1   0   0   1 255   0   0   0   0   1   0   0
    0   0]
 [  0   2   3   0   0   1   2   0   0   1   0 248   0   1   0   0

# RESULT
Also worse, but not by much (0.8843 down to 0.8727)

* Try training an svm with a linear kernel using binary vectors, count vectors, also apply a tfidf transform to the data. Compare performance with Naive Bayes.

In [ ]:
# from sklearn.svm import SVC
# from sklearn.feature_extraction.text import TfidfVectorizer

# # getting 20newsgroups dataset
# data = datasets.fetch_20newsgroups(subset='all',remove=('headers','footers','quotes'))

# data_train_X, data_test_X = train_test_split(data.data, random_state=1 )
# data_train_y, data_test_y = train_test_split(data.target, random_state=1 )

# vectorizer = TfidfVectorizer(binary=True)
# vectors_train=vectorizer.fit_transform(data_train_X)
# vectors_test=vectorizer.transform(data_test_X)

# svm = SVC(gamma='auto')
# svm.fit(vectors_train, data_train_y)

# #print a confusion matrix and accuracy
# print(confusion_matrix(svm.predict(vectors_test), data_test_y))
# print(accuracy_score(svm.predict(vectors_test), data_test_y))

# RESULT
* This actually took really long, and after researching a bit on the web it seems this is to be expected...
* I just let it keep running, i hope nothing bad happens

# Homework Problem 2
- Use the fetch_sms_spam() function below to get the sms spam data set.
- More info: [Here](https://archive.ics.uci.edu/ml/datasets/sms+spam+collection)
- Apply the vectorization and classification tasks above to  the data.
- Which gave the best performance (use a confusionmatrix and accuraccy score to help decide this).

In [ ]:
# It's worth while reading through this function there's useful things here, that
# I'm not explicitly covering in class.
def fetch_sms_spam():
  import requests # requests is a handy http library
  import zipfile # a zip library
  r=requests.get('http://www.dt.fee.unicamp.br/~tiago/smsspamcollection/smsspamcollection.zip')
  # I know of no way to get data out of a zipfile without saving to disk extracting and reading in, sigh
  zf=open('smsspam.zip','wb')
  zf.write(r.content)
  zip_ref = zipfile.ZipFile('smsspam.zip', 'r')
  zip_ref.extractall('smsspam')
  zip_ref.close()
  zf.close
  sms_file=open('smsspam/SMSSpamCollection.txt','r')
  #object to return
  data = {'data':[], 'target':[], 'target_classes': ['ham', 'spam']}
  # First tab splits the class and the SMS message
  # There's an argument to be made I should use the csv library and use
  # delimiter = '\t', I can't argue with that - it's a good idea. I did it this
  # way for pedogogy.
  for line in sms_file:
    idx = line.find('\t')
    target = line[:idx]
    doc = line[idx+1:]
    data['data'] += [doc]
    if target == 'ham': data['target'] += [0]
    else: data['target'] += [1]
  sms_file.close()
  return data

In [ ]:
data1 = fetch_sms_spam()

In [8]:
from sklearn.svm import SVC

data_train_X, data_test_X = train_test_split(data1['data'], random_state=1 )
data_train_y, data_test_y = train_test_split(data1['target'], random_state=1 )

vectorizer = CountVectorizer(binary=True)
vectors_train=vectorizer.fit_transform(data_train_X)
vectors_test=vectorizer.transform(data_test_X)

svm = SVC()
svm.fit(vectors_train, data_train_y)

#print a confusion matrix and accuracy
print(confusion_matrix(svm.predict(vectors_test), data_test_y))
print(accuracy_score(svm.predict(vectors_test), data_test_y))

/usr/local/lib/python3.6/dist-packages/sklearn/svm/base.py:196: FutureWarning: The default value of gamma will change from 'auto' to 'scale' in version 0.22 to account better for unscaled features. Set gamma explicitly to 'auto' or 'scale' to avoid this warning.
  "avoid this warning.", FutureWarning)


[[1212  182]
 [   0    0]]
0.8694404591104734


# RESULT

* Not entirely sure what to make of that...

* Accuracy score looks decent, at least.

In [9]:
from sklearn.svm import SVC

data_train_X, data_test_X = train_test_split(data1['data'], random_state=1 )
data_train_y, data_test_y = train_test_split(data1['target'], random_state=1 )

vectorizer = CountVectorizer(binary=False)
vectors_train=vectorizer.fit_transform(data_train_X)
vectors_test=vectorizer.transform(data_test_X)

svm = SVC()
svm.fit(vectors_train, data_train_y)

#print a confusion matrix and accuracy
print(confusion_matrix(svm.predict(vectors_test), data_test_y))
print(accuracy_score(svm.predict(vectors_test), data_test_y))

/usr/local/lib/python3.6/dist-packages/sklearn/svm/base.py:196: FutureWarning: The default value of gamma will change from 'auto' to 'scale' in version 0.22 to account better for unscaled features. Set gamma explicitly to 'auto' or 'scale' to avoid this warning.
  "avoid this warning.", FutureWarning)


[[1212  182]
 [   0    0]]
0.8694404591104734


# RESULT

* vectors being binary or not doesn't matter for SVC

In [10]:
data_train_X, data_test_X = train_test_split(data1['data'], random_state=1 )
data_train_y, data_test_y = train_test_split(data1['target'], random_state=1 )

vectorizer = CountVectorizer(binary=False)
vectors_train=vectorizer.fit_transform(data_train_X)
vectors_test=vectorizer.transform(data_test_X)

nb = MultinomialNB(alpha=.1)
nb.fit(vectors_train, data_train_y)

#print a confusion matrix and accuracy
print(confusion_matrix(nb.predict(vectors_test), data_test_y))
print(accuracy_score(nb.predict(vectors_test), data_test_y))

[[1204    7]
 [   8  175]]
0.9892395982783357


# RESULT
* Again, confusion matrix is a bit odd...
* But accuracy score is way better with Naive Bayes!!

In [11]:
vectorizer = CountVectorizer(binary=True)
vectors_train=vectorizer.fit_transform(data_train_X)
vectors_test=vectorizer.transform(data_test_X)

nb = MultinomialNB(alpha=.1)
nb.fit(vectors_train, data_train_y)

#print a confusion matrix and accuracy
print(confusion_matrix(nb.predict(vectors_test), data_test_y))
print(accuracy_score(nb.predict(vectors_test), data_test_y))

[[1203    7]
 [   9  175]]
0.9885222381635581


# RESULT
* nearly identical accuracy score no matter if binary or not

# CONCLUSION

* out of previous tests of various prediction models, it was found that the accuracy scores are as follow:
  * SVC = 0.8694
  * NB w/ Binary Vectors = 0.9885
  * NB w/o Binary Vectors = 0.9892
* It looks like the Naive Bayes without Binary Vectors performed the best for the spam detection data!
* The following code produces only the accuracy score for all the models created in this assignment

In [15]:
# -*- coding: utf-8 -*-
"""
Created on Sun Mar  3 14:43:55 2019

@author: matti
"""

# =============================================================================
# Naive Bayes with binary vectors as applied to the unclean 20newsgroups data
# =============================================================================
# getting 20newsgroups dataset
data = datasets.fetch_20newsgroups(subset='all')

data_train_X, data_test_X = train_test_split(data.data, random_state=1 )
data_train_y, data_test_y = train_test_split(data.target, random_state=1 )

# vectorizer = CountVectorizer(binary=False)
vectorizer = CountVectorizer(binary=True)
vectors_train=vectorizer.fit_transform(data_train_X)
vectors_test=vectorizer.transform(data_test_X)

nb = MultinomialNB(alpha=.1)
nb.fit(vectors_train, data_train_y)

#print the accuracy score
print('The accuracy score for the Naive Bayes with binary vectors as applied to the unclean 20newsgroups data is: ',accuracy_score(nb.predict(vectors_test), data_test_y))


# =============================================================================
# Naive Bayes with binary vectors as applied to the cleaned 20newsgroups data without headers, footers, and quotes
# =============================================================================
# getting cleaned data for training
data_clean = datasets.fetch_20newsgroups(subset='all',remove=('headers','footers','quotes'))

data_clean_train_X, data_test_X = train_test_split(data_clean.data, random_state=1 )
data_clean_train_y, data_test_y = train_test_split(data_clean.target, random_state=1 )

# vect_clean = CountVectorizer(binary=False)
vect_clean = CountVectorizer(binary=True)
vect_clean_train=vect_clean.fit_transform(data_train_X)
vect_clean_test=vect_clean.transform(data_test_X)

nb = MultinomialNB(alpha=.1)
nb.fit(vect_clean_train, data_train_y)

#print the accuracy score
print('The accuracy score for the Naive Bayes with binary vectors as applied to the cleaned 20newsgroups data without headers, footers, and quotes: ',accuracy_score(nb.predict(vect_clean_test), data_test_y))


# =============================================================================
# Naive Bayes without binary vectors as applied to the cleaned 20newsgroups data
# =============================================================================
data_clean = datasets.fetch_20newsgroups(subset='all',remove=('headers','footers','quotes'))

data_clean_train_X, data_test_X = train_test_split(data_clean.data, random_state=1 )
data_clean_train_y, data_test_y = train_test_split(data_clean.target, random_state=1 )

vect_false = CountVectorizer(binary=False)
vect_false_train=vect_false.fit_transform(data_train_X)
vect_false_test=vect_false.transform(data_test_X)

nb = MultinomialNB(alpha=.1)
nb.fit(vect_false_train, data_train_y)

#print the accuracy score
print('The accuracy score for the Naive Bayes without binary vectors as applied to the cleaned 20newsgroups data: ',accuracy_score(nb.predict(vect_false_test), data_test_y))


# =============================================================================
# Naive Bayes without binary vectors as applied to the unclean 20newsgroups data
# =============================================================================
# getting 20newsgroups dataset
data = datasets.fetch_20newsgroups(subset='all')

data_train_X, data_test_X = train_test_split(data.data, random_state=1 )
data_train_y, data_test_y = train_test_split(data.target, random_state=1 )

vectorizer = CountVectorizer(binary=False)
vectors_train=vectorizer.fit_transform(data_train_X)
vectors_test=vectorizer.transform(data_test_X)

nb = MultinomialNB(alpha=.1)
nb.fit(vectors_train, data_train_y)

#print the accuracy score
print('The accuracy score for the Naive Bayes without binary vectors as applied to the unclean 20newsgroups data: ',accuracy_score(nb.predict(vectors_test), data_test_y))


# =============================================================================
# The script is getting hung up on the SVC fitting...
# =============================================================================
#from sklearn.svm import SVC
#from sklearn.feature_extraction.text import TfidfVectorizer
#
## getting 20newsgroups dataset
#data = datasets.fetch_20newsgroups(subset='all',remove=('headers','footers','quotes'))
#
#data_train_X, data_test_X = train_test_split(data.data, random_state=1 )
#data_train_y, data_test_y = train_test_split(data.target, random_state=1 )
#
#vectorizer = TfidfVectorizer(binary=True)
#vectors_train=vectorizer.fit_transform(data_train_X)
#vectors_test=vectorizer.transform(data_test_X)
#
#svm = SVC(gamma='auto')
#svm.fit(vectors_train, data_train_y)
#
#
#print(confusion_matrix(svm.predict(vectors_test), data_test_y))
#print(accuracy_score(svm.predict(vectors_test), data_test_y))
# =============================================================================
# I will debug this later.
# =============================================================================

# =============================================================================
# Making the function for getting the sms spam dataset
# =============================================================================
# It's worth while reading through this function there's useful things here, that
# I'm not explicitly covering in class.
def fetch_sms_spam():
  import requests # requests is a handy http library
  import zipfile # a zip library
  r=requests.get('http://www.dt.fee.unicamp.br/~tiago/smsspamcollection/smsspamcollection.zip')
  # I know of no way to get data out of a zipfile without saving to disk extracting and reading in, sigh
  zf=open('smsspam.zip','wb')
  zf.write(r.content)
  zip_ref = zipfile.ZipFile('smsspam.zip', 'r')
  zip_ref.extractall('smsspam')
  zip_ref.close()
  zf.close
  sms_file=open('smsspam/SMSSpamCollection.txt','r')
  #object to return
  data = {'data':[], 'target':[], 'target_classes': ['ham', 'spam']}
  # First tab splits the class and the SMS message
  # There's an argument to be made I should use the csv library and use
  # delimiter = '\t', I can't argue with that - it's a good idea. I did it this
  # way for pedogogy.
  for line in sms_file:
    idx = line.find('\t')
    target = line[:idx]
    doc = line[idx+1:]
    data['data'] += [doc]
    if target == 'ham': data['target'] += [0]
    else: data['target'] += [1]
  sms_file.close()
  return data

data1 = fetch_sms_spam()


# =============================================================================
# SVC model with binary vectors as applied to the sms spam data
# =============================================================================
data_train_X, data_test_X = train_test_split(data1['data'], random_state=1 )
data_train_y, data_test_y = train_test_split(data1['target'], random_state=1 )

vectorizer = CountVectorizer(binary=True)
vectors_train=vectorizer.fit_transform(data_train_X)
vectors_test=vectorizer.transform(data_test_X)

svm = SVC(gamma='auto')
svm.fit(vectors_train, data_train_y)

#print the accuracy score
print('The accuracy score for the SVC model with binary vectors as applied to the sms spam data: ',accuracy_score(svm.predict(vectors_test), data_test_y))


# =============================================================================
# SVC model without binary vectors as applied to the sms spam data
# =============================================================================
from sklearn.svm import SVC

data_train_X, data_test_X = train_test_split(data1['data'], random_state=1 )
data_train_y, data_test_y = train_test_split(data1['target'], random_state=1 )

vectorizer = CountVectorizer(binary=False)
vectors_train=vectorizer.fit_transform(data_train_X)
vectors_test=vectorizer.transform(data_test_X)

svm = SVC(gamma='auto')
svm.fit(vectors_train, data_train_y)

#print the accuracy score
print('The accuracy score for the SVC model without binary vectors as applied to the sms spam data: ',accuracy_score(svm.predict(vectors_test), data_test_y))


# =============================================================================
# Naive Bayes model with binary vectors as applied to the sms spam data
# =============================================================================
data_train_X, data_test_X = train_test_split(data1['data'], random_state=1 )
data_train_y, data_test_y = train_test_split(data1['target'], random_state=1 )

vectorizer = CountVectorizer(binary=True)
vectors_train=vectorizer.fit_transform(data_train_X)
vectors_test=vectorizer.transform(data_test_X)

nb = MultinomialNB(alpha=.1)
nb.fit(vectors_train, data_train_y)

#print the accuracy score
print('The accuracy score for the Naive Bayes model with binary vectors as applied to the sms spam data: ',accuracy_score(nb.predict(vectors_test), data_test_y))


# =============================================================================
# Naive Bayes model without binary vectors as applied to the sms spam data
# =============================================================================
vectorizer = CountVectorizer(binary=False)
vectors_train=vectorizer.fit_transform(data_train_X)
vectors_test=vectorizer.transform(data_test_X)

nb = MultinomialNB(alpha=.1)
nb.fit(vectors_train, data_train_y)

#print the accuracy score
print('The accuracy score for the Naive Bayes model without binary vectors as applied to the sms spam data: ',accuracy_score(nb.predict(vectors_test), data_test_y))



The accuracy score for the Naive Bayes with binary vectors as applied to the unclean 20newsgroups data is:  0.8843378607809848
The accuracy score for the Naive Bayes with binary vectors as applied to the cleaned 20newsgroups data without headers, footers, and quotes:  0.7425721561969439
The accuracy score for the Naive Bayes without binary vectors as applied to the cleaned 20newsgroups data:  0.7292020373514432
The accuracy score for the Naive Bayes without binary vectors as applied to the unclean 20newsgroups data:  0.8726655348047538
The accuracy score for the SVC model with binary vectors as applied to the sms spam data:  0.8694404591104734
The accuracy score for the SVC model without binary vectors as applied to the sms spam data:  0.8694404591104734
The accuracy score for the Naive Bayes model with binary vectors as applied to the sms spam data:  0.9885222381635581
The accuracy score for the Naive Bayes model without binary vectors as applied to the sms spam data:  0.9892395982783